dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -130,39 +130,42 @@ def _add_helper(self, other1, other2):
         """
-        # Merge Variance
-        self.variance = self._merge_variance(
-            other1.match_count, other1.variance, other1.mean,
-            other2.match_count, other2.variance, other2.mean)
-
-        # Merge min, max, sum and histograms
-        if other1.min is not None and other2.min is not None:
-            # update histogram
-            self._add_helper_merge_profile_histograms(other1, other2)
-
-            # update min, max, sum
-            self.min = min(other1.min, other2.min)
-            self.max = max(other1.max, other2.max)
-            self.sum = other1.sum + other2.sum
-
-        elif other2.min is None:
-            # update histogram
-            self.histogram_methods = other1.histogram_methods
-            self.quantiles = other1.quantiles
-
-            # update min, max, sum
-            self.min = other1.min
-            self.max = other1.max
-            self.sum = other1.sum
-        else:
-            # update histogram
-            self.histogram_methods = other2.histogram_methods
-            self.quantiles = other2.quantiles
-
-            # update min, max, sum
-            self.min = other2.min
-            self.max = other2.max
-            self.sum = other2.sum
+        # Merge variance, histogram, min, max, and sum
+        if "variance" in self.__calculations.keys():
+            self.variance = self._merge_variance(
+                other1.match_count, other1.variance, other1.mean,
+                other2.match_count, other2.variance, other2.mean)
+        if "histogram_and_quantiles" in self.__calculations.keys():
+            if other1.histogram_selection != None and \
+                    other2.histogram_selection != None:
+                self._add_helper_merge_profile_histograms(other1, other2)
+            elif other2.histogram_selection == None:
+                self.histogram_methods = other1.histogram_methods
+                self.quantiles = other1.quantiles
+            else:
+                self.histogram_methods = other2.histogram_methods
+                self.quantiles = other2.quantiles
+        if "min" in self.__calculations.keys():
+            if other1.min != None and other2.min != None:
+                self.min = min(other1.min, other2.min)
+            elif other2.min == None:
+                self.min = other1.min
+            else:
+                self.min = other2.min
+        if "max" in self.__calculations.keys():
+            if other1.max != None and other2.max != None:
+                self.max = max(other1.max, other2.max)
+            elif other2.max == None:
+                self.max = other1.max
+            else:
+                self.max = other2.max
+        if "sum" in self.__calculations.keys():
+            if other1.sum != 0 and other2.sum != 0:
+                self.sum = other1.sum + other2.sum
+            elif other2.sum == 0:
+                self.sum = other1.sum
+            else:
+                self.sum = other2.sum
```

Comment:
Are these 3 separate cases necessary? If you case on whether one or the other is zero, so that you can just use the other sum, then thats the same as the first case when you add them, so you can just always add them like in the first case. Right?

Reply:
Yep, good catch, fixed